packages/web/pages/dashboard.tsx

```diff
@@ -0,0 +1,258 @@
+            color="purple.400"
+            onClick={handleReset}
+            leftIcon={<DeleteIcon />}
+          >
+            Reset
+          </MetaButton>
+        )}
+        <MetaButton
+          aria-label="Edit layout"
+          colorScheme="purple"
+          textTransform="uppercase"
+          px={12}
+          letterSpacing="0.1em"
+          size="lg"
+          fontSize="sm"
+          bg="transparent"
+          color={editable ? 'red.400' : 'purple.400'}
+          leftIcon={<EditIcon />}
+          transition="color 0.2s ease"
+          onClick={toggleEditLayout}
+        >
+          Edit layout
+        </MetaButton>
+      </ButtonGroup>
+
+      <ResponsiveGridLayout
+        className="grid"
+        onLayoutChange={(layout, layouts) => {
+          setCurrent({ layout, layouts });
+          setChanged(true);
+        }}
+        verticalCompact
+        layouts={gridLayouts}
+        breakpoints={{ xl: 1920, lg: 1180, md: 900, sm: 768, xs: 480, xxs: 0 }}
+        preventCollision={false}
+        cols={{ xl: 12, lg: 12, md: 12, sm: 4, xs: 4, xxs: 4 }}
+        rowHeight={135}
+        autoSize
+        // isBounded
+        isDraggable={!!editable}
+        isResizable={!!editable}
+        onDragStart={toggleScrollLock}
+        onDragStop={toggleScrollLock}
+        onResizeStart={toggleScrollLock}
+        onResizeStop={toggleScrollLock}
+        transformScale={1}
+        margin={{
+          xl: [20, 20],
+          lg: [20, 20],
+          md: [20, 20],
+          sm: [20, 20],
+          xs: [20, 20],
+          xxs: [20, 20],
+        }}
+        containerPadding={{
+          xl: [20, 20],
+          lg: [20, 20],
+          md: [15, 15],
+          sm: [15, 15],
+          xs: [10, 10],
+          xxs: [10, 10],
+        }}
+      >
+        {/* <DashboardSection key="latest" id="latest" containerQuery={queryData}> */}
+        <Box key="latest" className="gridItem">
+          <GridItem title="Latest Content" sx={gridConfig.latest}>
+            <LatestContent />
+          </GridItem>
+        </Box>
+        <Box key="xp" className="gridItem">
+          <GridItem title="XP" sx={gridConfig.xp}>
+            <XP />
+          </GridItem>
+        </Box>
+        <Box key="seed" className="gridItem">
+          <GridItem title="Seed" sx={gridConfig.seed}>
+            <Seed />
+          </GridItem>
+        </Box>
+        <Box key="calendar" className="gridItem">
+          <GridItem title="Calendar" sx={gridConfig.calendar}>
+            <Calendar />
+          </GridItem>
+        </Box>
+        <Box key="leaderboard" className="gridItem">
+          <GridItem title="Leaderboard" sx={gridConfig.leaderboard}>
+            <Leaderboard />
+          </GridItem>
+        </Box>
+      </ResponsiveGridLayout>
+    </Box>
+  );
+};
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export function resetLayouts() {
+  if (getFromLS('metagame-dashboard') === null) {
```

Comment:
This will try to retrieve `localStorage['metagame-dashboard']['metagame-dashboard']` and will be `undefined`.

Reply:
> @Udit-takkar @luxumbra let's read through dysbullic's comments and fix what we can fix in our respective sections that we worked on.

I think most of the review is focused on my dashboard code, rather than Udits XP section. Do you mean fix in this PR or in our respective PRs?